lib/blockscore/collection.rb

```diff
@@ -1,8 +1,48 @@
 module BlockScore
   class Collection < Array
-    def initialize(target)
+
+    attr_reader :default_params
+    protected :default_params
+
+    attr_reader :parent, :target, :data
+
+
+    def initialize(params)
       super()
-      @target = target
+      @parent = params.fetch :parent
+      @target = target.fetch :target
+      @default_params = parent ? { "#{parent_name}_id": parent.id } : {}
+      register_parent_data
+    end
+
+    def all
+      self
+    end
+
+    def inspect
+      "#<#{self.class}:0x#{object_id.to_s(16)} LAZY:#{ Util.to_plural(target_name) } {data}>"
+    end
+
+    def parent_name
+      parent.class.resource
+    end
+
+    def target_name
+      target.class.resource
+    end
+
+    def create(params)
+      assoc_params = default_params.merge(params)
+      item = target.create assoc_params
+      @data << item.id
+      self << item
+      item
+    end
+
+    def retrieve(id)
+      if data.include? id
+        self[data.index(id)]
```

Comment:
Style/IfUnlessModifier: Favor modifier `if` usage when having a single-line body. Another good alternative is the usage of control flow `&&`/`||`.